Fixed rouding problem on calc of spherical law of cosines.

dusiema opened the pull request:
I had the same calculation in one of my applications and my test colleague found a bug in it. After investigation we found that there was a precision problem wiht the double values that lead to values greater 1 and then to NaN as a result of calling `acos` on it.

I came across your lib while looking for a more robust solution but then found the same "bug". 

So here is proposal to overcome this rounding problem.

gr211 commented:
//spherical law of cosines
double sphereCos = (sin(slat) * sin(flat)) + (cos(slat) * cos(flat) * cos(diffLongitudes));
double c = acos(max(min(sphereCos, 1d), -1d));

dusiema commented:
Yes, no problem. I changed it.

One more thing: While I looked at the pull request again I realized I also changed the order of params in `com.grum.geocalc.DistanceTest.testZeroDistance()`.
